subsys/bluetooth/controller/Kconfig

```diff
@@ -293,7 +379,7 @@ config BT_CTLR_RL_SIZE
 
 config BT_CTLR_EXT_SCAN_FP
 	bool "LE Extended Scanner Filter Policies"
-	depends on BT_OBSERVER
+	depends on BT_OBSERVER && BT_CTLR_EXT_SCAN_FP_SUPPORT
 	default y
 	help
 	  Enable support for Bluetooth v4.2 LE Extended Scanner Filter Policies
```

Comment:
I can't comment on lines not part of the diff ...

```
config BT_CTLR_DATA_LENGTH
	# Hidden option to enable support for Bluetooth v4.2 LE Data Length
	# Update procedure in the Controller.
	bool
	depends on BT_DATA_LEN_UPDATE && BT_CTLR_DATA_LENGTH_UPDATE_SUPPORT
	default y
```

and

```
config BT_CTLR_PHY
	# Hidden option to enable support for Bluetooth 5.0 PHY Update
	# Procedure in the Controller.
	bool
	depends on BT_PHY_UPDATE && BT_CTLR_PHY_UPDATE_SUPPORT
	select BT_CTLR_EXT_REJ_IND
	default y
```

We might also need `_PHY_2M_SUPPORTED` and `_PHY_CODED_SUPPORTED`

Reply:
> Sorry to add these comments so late. I missed it earlier.

That's OK, they're not really hard to address. As such, I'll run our internal regression suite and then re-push to ensure that I haven't broken something else.

Anyhow, for the 2M_SUPPORTED and CODED_SUPPORTED, I think that should be addressed in a separate PR, does this make sense to you?